src/featbin/wav-reverberate.cc

```diff
@@ -0,0 +1,325 @@
+// featbin/extract-segments.cc
+
+// Copyright 2015  Tom Ko
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "feat/wave-reader.h"
+
+namespace kaldi {
+
+void ElementwiseProductOfFft(const Vector<BaseFloat> &a, Vector<BaseFloat> *b) {
+  for (int i = 0; i < a.Dim() / 2; i++) {
+    float tmp = (*b)(2*i);
+    (*b)(2*i) = tmp * a(2*i) - (*b)(2*i+1) * a(2*i+1);
+    (*b)(2*i+1) = tmp * a(2*i+1) + (*b)(2*i+1) * a(2*i);
+  }
+}
+
+void ConvolveSignals(const Vector<BaseFloat> &filter, Vector<BaseFloat> *signal) {
+  int signal_length = signal->Dim();
+  int filter_length = filter.Dim();
+
+  int fft_length = 2;
+  while (fft_length < signal_length + filter_length - 1) {
+    fft_length = fft_length * 2;
+  }
+  KALDI_VLOG(1) << "fft_length for full signal convolution is " << fft_length;
+
+  SplitRadixRealFft<BaseFloat> *srfft_;
+  srfft_ = new SplitRadixRealFft<BaseFloat>(fft_length);
```

Comment:
You are right, it should just be
SplitRadixRealFft<BaseFloat> srfft(fft_length);
[in case "stack allocation" was not clear to Tom.]